contracts/VotingEscrow.vy

```diff
@@ -1,268 +1,135 @@
+@dev
+    Vote weight decays linearly over time. Lock time cannot be more than 4 years.
+    A user can unlock funds early incurring a penalty.
 """
-
-# Voting escrow to have time-weighted votes
-# Votes have a weight depending on time, so that users are committed
-# to the future of (whatever they are voting for).
-# The weight in this implementation is linear, and lock cannot be more than maxtime:
-# w ^
-# 1 +        /
-#   |      /
-#   |    /
-#   |  /
-#   |/
-# 0 +--------+------> time
-#       maxtime (4 years?)
+from vyper.interfaces import ERC20
 
 struct Point:
     bias: int128
     slope: int128  # - dweight / dt
     ts: uint256
     blk: uint256  # block
-# We cannot really do block numbers per se b/c slope is per time, not per block
-# and per block could be fairly bad b/c Ethereum changes blocktimes.
-# What we can do is to extrapolate ***At functions
 
 struct LockedBalance:
-    amount: int128
+    amount: uint256
     end: uint256
 
-
-interface ERC20:
-    def decimals() -> uint256: view
-    def name() -> String[64]: view
-    def symbol() -> String[32]: view
-    def transfer(to: address, amount: uint256) -> bool: nonpayable
-    def transferFrom(spender: address, to: address, amount: uint256) -> bool: nonpayable
-    def approve(spender: address, amount: uint256) -> bool: nonpayable
-
 interface IVeYfiRewards:
-    def queueNewRewards(_amount: uint256) -> bool: nonpayable
-
-interface Migrator:
-    def migrateLock(_account: address, _amount: uint256):nonpayable
-
-DEPOSIT_FOR_TYPE: constant(int128) = 0
-CREATE_LOCK_TYPE: constant(int128) = 1
-INCREASE_LOCK_AMOUNT: constant(int128) = 2
-INCREASE_UNLOCK_TIME: constant(int128) = 3
-
-
-event CommitOwnership:
-    admin: address
-
-event CommitUnlocker:
-    unlocker: address
-
-event ApplyOwnership:
-    admin: address
-
-event ApplyUnlocker:
-    unlocker: address
+    def queueNewRewards(amount: uint256) -> bool: nonpayable
 
 event Deposit:
-    deposit_from: indexed(address)
-    provider: indexed(address)
-    value: uint256
-    locktime: indexed(uint256)
-    type: int128
+    sender: indexed(address)
+    user: indexed(address)
+    action: indexed(uint256)
+    amount: uint256
+    locktime: uint256
     ts: uint256
 
 event Withdraw:
-    provider: indexed(address)
-    value: uint256
+    user: indexed(address)
+    amount: uint256
+    ts: uint256
+
+event Penalty:
+    user: indexed(address)
+    amount: uint256
     ts: uint256
 
 event Supply:
-    prevSupply: uint256
-    supply: uint256
+    old_supply: uint256
+    new_supply: uint256
+    ts: uint256
 
 event Initialized:
     token: address
+    reward_pool: address
 
-event NewRewardPool:
-    reward_pool: indexed(address)
-
-event Migrate:
-    account: indexed(address)
-    amount: uint256
-    to: indexed(address)
-
-event NextVeContractSet:
-    ve: indexed(address)
-
-event QueuedNextVeContractSet:
-    ve: indexed(address)
+# enum DepositAction
```

Comment:
Since enum is comming to vyper, and enums are going to be integer values of 2**n the numbers should also be powers of two.

Reply:
fixed in https://github.com/yearn/veYFI/pull/160/commits/eced624b23d6d95dfbbf251a7029aeb0973e3a62